include/rocksdb/table.h

```diff
@@ -342,6 +342,20 @@ struct BlockBasedTableOptions {
   // This must generally be true for gets to be efficient.
   bool whole_key_filtering = true;
 
+  // If true, the memory usage of (new) Bloom filter and Ribbon
+  // filter construction will be reserved by charging to the block cache
+  // associated with the block based table builder.
+  //
+  // It helps account the memory usage of such filter construction
+  // and constrain the memory usage of constructing
+  // the banding in Ribbon filter, by switching to Bloom filter
+  // if that banding portion uses up too much memory relative to
+  // the avaible space left in the block cache
+  // (i.e, causing a cache full when strict_capacity_limit = true)
+  //
+  // Default: false
+  bool reserve_bloom_ribbon_filter_construction_memory = false;
```

Comment:
I prefer simple: one omnibus option for whether to reserve in block cache our best estimate of any non-negligible memory for table construction. I would not assume "everyone" will eventually want to charge everything, because some people use a small block cache because they do not prioritize read performance.

Reply:
> one omnibus option for whether to reserve in block cache our best estimate of any non-negligible memory for table construction

Should such an option include table reader memory too? Or would that be a separate option in `BlockBasedTableOptions`? 

I am inclined toward bundling and not offering ability to disable it for new features. People with small block cache who don't care about reads and have strict_capacity_limit == 0 likely won't notice. Those who will notice have a chance to notice (and adjust their block cache in response) when they read about/try out the new feature, so hopefully we can minimize their chance of only noticing once in production. 

But I don't care that much. Having one option for charging known table reader memory and another option for charging known table builder memory is fine with me.

Reply:
Discussed offline: use `reserve_table_builder_mem`